roles/splunk/tasks/configure_license.yml

```diff
@@ -1,4 +1,33 @@
 ---
+- block:
+  - name: Copy Licenses to Master
+    copy:
+      src: "{{ item }}"
+      dest: "/tmp/{{ item }}"
```

Comment:
Remove `/tmp/` as (1) it may not exist and (2) users can already enter a path when defining `splunk_licenses` so hardcoding it here isn't required

Reply:
In the current state, user is defining a source, so it has to exists on the ansible host (source) and on the licensemaster (target). Maybe that's not guaranteed at all.
So, what about to define splunk_licenses as hash table, so user can define where to store them temporary?

Reply:
You could just use two variables, one for `splunk_license_ansible_path` and one for `splunk_license_target_path`

Reply:
I think I'd agree with having the license be defined as a text blob in a variable, and installed via [the REST API](https://docs.splunk.com/Documentation/SplunkCloud/8.1.2103/RESTREF/RESTlicense#licenser.2Flicenses).  The `add licenses` command used later in this PR already requires auth, so there's nothing holding you back from making the change via authenticated REST API.

Reply:
The difficulty may be in doing that idempotently, which is a requirement when working with Ansible.  The REST API returns the hash of the license, which is then used to query the REST API for the license details later.  What's not clear is if you could precalculate that, or if it's in the `.lic` file already (I didn't see it, but did see a `GUID`).

Reply:
Regarding `/tmp`, you can create a tempfile/directory in ansible using builtin methods: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/tempfile_module.html

Reply:
@schneewe We think the best way to implement this functionality would actually be to use the `uri` module to send an HTTP POST to the API with the license in the payload. See: https://docs.splunk.com/Documentation/Splunk/8.1.3/RESTREF/RESTlicense#licenser.2Flicenses

You can reuse some of the parameters from https://github.com/splunk/ansible-role-for-splunk/blob/5163490b765a85c334fce1f4d6e316dbdb0f87f2/roles/splunk/tasks/install_apps.yml#L45  to handle authentication.

Reply:
To handle multiple licenses, the `splunk_licenses` var would contain a list of file paths leveraging the file_contents lookup. Then, use a loop with `splunk_licenses` in the `uri` module to set the payload.

Reply:
I get your point and completely agree, but I think about the very first run, this would not be possible because splunk isn't running therefore no API available. If splunk is already installed, this is the best solution.
@micahkemp the way with the temp folder is very nice, didn't know that.

Reply:
@schneewe is there a reason the licenses need to be configured before first start of splunk?

Reply:
@micahkemp-splunk  I think with the current proposed implementation, it would be an order of operations thing, since configure_license.yml is called during the install process (since it previously only supported LM clients). That said, the easy way to fix that would be to just move these new tasks to their own file, like `configure_lm.yml` which could then be called as the next play after installation.